### DISCUSSION TITLE
Copy and rename a file instead of moving it

### DISCUSSION THREAD

**Discussion by GNeulaender** [SilentVoid13/Templater#1206] [Help] [answered]
I'm new to Templater (and Obsidian as a whole) and have been struggling to workout the tp.obsidian commands.

Is there a way to copy a file from one folder to another instead of moving it?

**Comment by Zachatoo** [comment]
You can use [tp.file.create_new](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilecreate_newtemplate-tfile--string-filename-string-open_new-boolean--false-folder-tfolder) and give it the current file's contents ([tp.file.content](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilecontent)) to do this.

```js
<%*
const folder = app.vault.getAbstractFileByPath("FOLDERNAME");
await tp.file.create_new(tp.file.content, "FILENAME", false, folder);
-%>
```

**Reply by GNeulaender** [reply]
I don't think it would fit my needs. I actually want to copy and rename a PDF file, having a version per node.

**Reply by GNeulaender** [reply, marked as answer]
I do think now I found a solution using some of your code. Thanks for the help!

```javascript
<%*
const reference_file = app.vault.getAbstractFileByPath("FILEPATH");
const current_file = tp.file.path(true);
const new_file = file.substring(0,file.length - 3) + ".pdf";
await app.vault.copy(reference_file, new_file);
-%>
```
